testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/jms/AbstractCreateJMSBridgeSetupTask.java

```diff
@@ -38,7 +38,7 @@ public void setup(ManagementClient managementClient, String containerId) throws
         jmsBridgeAttributes.get("source-destination").set(QUEUE1_JNDI_NAME);
         jmsBridgeAttributes.get("target-connection-factory").set(CF_JNDI_NAME);
         jmsBridgeAttributes.get("target-destination").set(QUEUE2_JNDI_NAME);
-        jmsBridgeAttributes.get("module").set("org.apache.activemq.artemis:main");
+        jmsBridgeAttributes.get("module").set("org.apache.activemq.artemis");
```

Comment:
@soul2zimate Was this change required to get the test to pass, or did you do this more as a bit of tidying?

I'm concerned this indicates we've introduced a breaking change. AIUI in the past appending ":main" was not necessary but wasn't wrong. If the test breaks without this change that indicates it is now "wrong", which is an incompatible change.

@ehsavoie @ropalka @dmlloyd FYI.

Reply:
+1 the module name should allow for a slot in that same pattern.

Reply:
Yes, some JMS tests in smoke break without this change.

Reply:
This is due to https://github.com/wildfly/wildfly-core/blob/a62fb1c4ab771bac5790b2eb2d369af39eec3e45/server/src/main/java/org/jboss/as/server/deployment/module/ModuleDependency.java#L151-L153. The simple fix is to use `ModuleIdentifier.fromString()`. However, I don't know if we simply just want a PR for that or not.

Reply:
I'll file a fix for this later today. FWIW we should add the `:main` back even if it's failing and add the core-upgrade-required label as we'll need a core upgrade after the fix is in.

Reply:
@jamezp I don't think ModuleDependency is relevant here. This PR isn't touching deployment processing code; it's all subsystem-level messaging container resources. The attribute in this test is used to create a subsystem resource, not to set up a ModuleDependency.

I think we'll likely need our own ModuleIdentifier from WF Core. ModuleLoader.loadModuleLocal is using the provided 'name' as is and doesn't try to recognize that "foo:main" and "foo" are equivalent. It was ModuleIdentifier that dealt with the fact that "foo" is the canonical representation of "foo:main". By running strings through it we got canonical strings.

Reply:
Yes I remember hitting those kind of issues while working on something else, I didn't dig them, sorry

Reply:
Note that in the https://github.com/wildfly/wildfly-core/pull/6270 I added this bit:

https://github.com/wildfly/wildfly-core/pull/6270/files#diff-867089bef7659adb3ead7e82d208a38aa95d3fda6df7476187f2aa272bf064d4R94-R110

Any AttributeDefinition that represents a module name can add that ParameterCorrector and that will ensure that what gets stored in the model is the correct form. If this 'module' attribute did that and this test left the "org.apache.activemq.artemis:main" value, that would demonstrate it works.